http/auth/adaptor.go

```diff
@@ -0,0 +1,128 @@
+package auth
+
+import (
+	"errors"
+	"github.com/casbin/casbin/v2/model"
+	"github.com/casbin/casbin/v2/persist"
+	"gorm.io/gorm"
+)
+
+const (
+	permissionTag = "p"
+	groupTag      = "g"
+)
+
+// adapter is a customized read-only gorm adapter for `Casbin`.
+// It can load policy from db
+type adapter struct {
+	db *gorm.DB
+}
+
+func (a *adapter) UpdatePolicy(string, string, []string, []string) error {
+	return errors.New("not implemented")
+}
+
+func (a *adapter) UpdatePolicies(string, string, [][]string, [][]string) error {
+	return errors.New("not implemented")
+}
+
+func (a *adapter) UpdateFilteredPolicies(string, string, [][]string, int, ...string) ([][]string, error) {
+	return nil, errors.New("not implemented")
+}
+
+// newAdapter is the constructor for adapter.
+func newAdapter(db *gorm.DB) *adapter {
```

Comment:
shift this constructor above (to below the type declaration) and so that all the not implemented methods are together?

Reply:
i mean shift this constructor above like in the auth.go